panther_lights/src/controller_node.py

```diff
@@ -111,43 +177,59 @@ def _percent_to_apa_driver_brightness(self, brightness_percent) -> int:
         raise ValueError('Brightness out of range <0,1>')
 
     def _controller_timer_cb(self, *args) -> None:
-        with self._lock:
-            if self._animation_finished or self._interrupt:
-                if self._anim_queue.empty():
-                    self._current_animation = self._default_animation
-                else:
-                    self._current_animation = self._anim_queue.get(block=False)
-                    self._interrupt = False
-
-            if self._current_animation:
-                if not self._current_animation.front.finished:
-                    frame_front = self._current_animation.front()
-                    brightness_front = self._current_animation.front.brightness
-                    self._controller.set_panel_frame(
-                        LightsControllerNode.PANEL_FRONT, frame_front, brightness_front
-                    )
-                if not self._current_animation.rear.finished:
-                    frame_rear = self._current_animation.rear()
-                    brightness_rear = self._current_animation.rear.brightness
-                    self._controller.set_panel_frame(
-                        LightsControllerNode.PANEL_REAR, frame_rear, brightness_rear
-                    )
-
-                self._animation_finished = (
-                    self._current_animation.front.finished and self._current_animation.rear.finished
+        if self._animation_finished:
+            self._anim_queue.validate_queue()
+
+            if self._default_animation and not self._anim_queue.has_animation(
+                self._default_animation
+            ):
+                self._default_animation.reset_time()
+                self._anim_queue.put(self._default_animation)
+
+            if not self._anim_queue.empty():
+                self._current_animation = self._anim_queue.get()
+
+        if self._current_animation:
+            if self._current_animation.priority > self._anim_queue.get_first_anim_priority():
+                if (
+                    self._current_animation.front.progress < 0.9
+                    and not self._current_animation.repeating
+                ):
+                    self._anim_queue.put(self._current_animation)
+                self._current_animation.front.reset()
+                self._current_animation.rear.reset()
+                self._animation_finished = True
+                self._current_animation = None
```

Comment:
```suggestion
                else:
                    del self._current_animation
                    self._current_animation = None
```
Since we know we won't need this object in the future, and it is a copy of the original object, we can just explicitly delete it and not care about resetting it